code/modules/atmospherics/machinery/components/components_base.dm

```diff
@@ -73,18 +73,19 @@
 // Pipenet stuff; housekeeping
 
 /obj/machinery/atmospherics/components/nullifyNode(i)
+	// Every node has a parent pipeline and an air associated with it.
 	if(parents[i])
 		nullifyPipenet(parents[i])
-		QDEL_NULL(airs[i])
-	..()
+	QDEL_NULL(airs[i])
+	return ..()
```

Comment:
why QDEL just airs, aren't the lists of gasmixtures meant to be synced to the list of parents?

Reply:
1. `airs[i]` are *stationary*, so they are meant to stay no regardless of node connection, should never be deleted without caution, and all of them should be deleted when a component gets deleted. The `airs[i]` are added to and removed from pipenets' `other_airs` to be `reconcile_air()`ed as needed, and that's the reason why we need to take `other_airs` list housekeeping seriously.
2. `nullifyPipenet(parents[i])` should remove other references to the qdel()ing gasmixtures.
3. Parentless gasmixtures *might* naturally GC without qdel() but it isn't expected code-wise that every parentless gasmixture will; there is `/datum/proc/Destory()`.